src/Google.Api.Gax/ApiCall.cs

```diff
@@ -6,54 +6,75 @@
  */
 
 using Google.Protobuf;
+using Grpc.Core;
+using System;
 using System.Threading.Tasks;
 
 namespace Google.Api.Gax
 {
-    /// <summary>
-    /// Wrapper for matching asynchronous and synchronous API calls to the same underlying method.
-    /// </summary>
-    /// <typeparam name="TRequest">Request type, which must be a protobuf message.</typeparam>
-    /// <typeparam name="TResponse">Response type, which must be a protobuf message.</typeparam>
+    public static class ApiCall
+    {
+        internal static ApiCall<TRequest, TResponse> Create<TRequest, TResponse>(
+            Func<TRequest, CallOptions, AsyncUnaryCall<TResponse>> asyncGrpcCall,
+            Func<TRequest, CallOptions, TResponse> syncGrpcCall,
+            CallSettings baseCallSettings,
+            IClock clock)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            return new ApiCall<TRequest, TResponse>(
+                asyncGrpcCall.WithTaskTransform().MapArg((CallSettings cs) => cs.ToCallOptions(clock)),
+                syncGrpcCall.MapArg((CallSettings cs) => cs.ToCallOptions(clock)),
+                baseCallSettings);
+        }
+    }
+
     public sealed class ApiCall<TRequest, TResponse>
         where TRequest : class, IMessage<TRequest>
         where TResponse : class, IMessage<TResponse>
     {
-        /// <summary>
-        /// Delegate representing a single asynchronous API call with a request and <see cref="CallSettings"/>.
-        /// </summary>
-        /// <param name="request">Request to make to the service.</param>
-        /// <param name="callSettings">If not null, provides settings that override the client-provided settings.</param>
-        /// <returns>A task representing the API call; the result of the completed task will be the API response.</returns>
-        public delegate Task<TResponse> AsyncCall(TRequest request, CallSettings callSettings);
-        /// <summary>
-        /// Delegate representing a single synchronous API call with a request and <see cref="CallSettings"/>.
-        /// </summary>
-        /// <param name="request">Request to make to the service.</param>
-        /// <param name="callSettings">If not null, provides settings that override the client-provided settings.</param>
-        /// <returns>The API response.</returns>
-        public delegate TResponse SyncCall(TRequest request, CallSettings callSettings);
-
-        /// <summary>
-        /// Initializes a new <see cref="ApiCall"/> with the specified asynchronous and synchronous calls.
-        /// </summary>
-        /// <param name="asyncCall">The asynchronous API call. Must not be null.</param>
-        /// <param name="syncCall">The synchronous API call. Must not be null.</param>
-        /// <exception cref="ArgumentNullException">Either specified call is null.</exception>
-        public ApiCall(AsyncCall asyncCall, SyncCall syncCall)
+        internal ApiCall(
```

Comment:
Thought for another round of refactoring - maybe we should call this `ApiOperation` or even `ApiMethod`, then CallSync and CallAsync methods, so they're verbs again. Not for now though :)